CMakeLists.txt

```diff
@@ -467,12 +468,16 @@ if(BUILD_LLVM_DWARF)
   target_link_libraries(binaryen llvm_dwarf)
 endif()
 
-if(MIMALLOC_STATIC)
-  if(NOT(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND BUILD_STATIC_LIB) OR EMSCRIPTEN)
+if(BUILD_MIMALLOC)
+  if(NOT(CMAKE_SYSTEM_NAME STREQUAL "Linux") OR EMSCRIPTEN)
     message(FATAL_ERROR "Statically linking mimalloc is only supported when building as a native, statically linked library on Linux.")
```

Comment:
Error message still references static library/static linking. Probably something like "Linking against mimalloc (statically or dynamically) is only supported for native Linux builds."